java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -429,6 +430,16 @@ public String toString() {
     public ClassTree declaration() {
       return declaration;
     }
+
+    public TypeSubstitution identitySubstitution() {
+      if (identitySubstitution == null) {
```

Comment:
not necessarily much better but at least worth sharing : http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Maps.html#toMap(java.lang.Iterable,%20com.google.common.base.Function)